macosx/Torrent.mm

```diff
@@ -429,6 +446,18 @@ - (NSString*)magnetLink
     return @(tr_torrentGetMagnetLink(fHandle));
 }
 
+- (NSString*)magnetLocation
+{
+    if (self.isMagnet)
+    {
+        if (tr_torrentGetDownloadDir(fHandle) != NULL)
+        {
+            return @(tr_torrentGetDownloadDir(fHandle));
+        }
+    }
+    return @"";
```

Comment:
It makes more sense to me to return `nil` here rather than an invalid value

Reply:
storing nil in an NSDictionary will cause a crash, so it makes sense to return an empty string

Reply:
Usually one would store an `NSNull` instance in a dictionary to indicate an empty value, or store nothing for this key at all

Reply:
ok sure, I thought it would be more readable this way

Reply:
ok sure, ive updated self.magnetLocation to now return nil,
and also added conditionals to the history dict.

It does add a bunch more lines of code,
but the archived history dict size will be smaller